pkg/controllers/user/helm/controller.go

```diff
@@ -166,12 +174,14 @@ func (l *Lifecycle) DeployApp(obj *v3.App) (*v3.App, error) {
 		}
 	}
 	newObj, err := v3.AppConditionInstalled.Do(obj, func() (runtime.Object, error) {
-		template, notes, tempDir, err := generateTemplates(obj, l.TemplateVersionClient, l.TemplateContentClient)
-		defer os.RemoveAll(tempDir)
+		template, notes, appDir, tempDir, err := l.generateTemplates(obj)
+		if tempDir != "" {
```

Comment:
Shouldn't really matter if dir is empty or not.

Reply:
Wasn't sure of the behavior, but tested to verify that nothing is removed for empty string, thanks for pointing this out!